src/MongoDB.Driver.Core/Core/Compression/SnappyCompressor.cs

```diff
@@ -27,28 +27,89 @@ internal class SnappyCompressor : ICompressor
     {
         public CompressorType Type => CompressorType.Snappy;
 
+        /* Declared in header order as defined in the spec:
+         * https://github.com/mongodb/specifications/blob/7461d48df580b2e32b33fdc01188c86fc79773c1/source/compression/OP_COMPRESSED.rst#op_compressed*/
+        private const int MessageLengthSize = sizeof(int);
+        private const int RequestIdSize = sizeof(int);
+        private const int ResponseToSize = sizeof(int);
+        private const int OpCodeSize = sizeof(int);
+        private const int OriginalOpCodeSize = OpCodeSize;
+        private const int UncompressedSizeSize = sizeof(int);
+        private const int CompressorTypeSize = sizeof(byte); // compressorId in spec
```

Comment:
These constants might no longer be needed depending on what changes you make below.